linux/vhba_hot_add_remove/wait_device_list_changed.yml

```diff
@@ -81,17 +91,13 @@
                 - guest_ansible_device[wait_device_name] is defined
                 - guest_ansible_device[wait_device_name].size is defined
                 - guest_ansible_device[wait_device_name].size == "0.00 Bytes"
-          when: wait_device_state | lower == 'absent'
-      when: >
-        ('Flatcar' in guest_os_ansible_distribution or
-         (guest_os_ansible_distribution == "Ubuntu" and
-          guest_os_ansible_distribution_major_ver | int >= 22) or
-         (guest_os_ansible_distribution == "Debian" and
-          guest_os_ansible_distribution_major_ver | int >= 12) or
-         guest_os_ansible_distribution == "Fedora")
-  when: new_disk_ctrl_type == 'lsilogic'
 
 - name: "Handle NVMe disk"
+  when:
+    - new_disk_ctrl_type == 'nvme'
+    - vm_hardware_version_num | int < 19
+    - guest_os_ansible_distribution != 'VMware Photon OS'
+    - guest_os_ansible_distribution != "FreeBSD"
```

Comment:
Line 99 ~ 100 can be merged into `guest_os_ansible_distribution not in ['VMware Photon OS', 'FreeBSD']`